web_src/js/markup/anchors.js

```diff
@@ -30,6 +30,20 @@ export function initMarkupAnchors() {
     });
     heading.prepend(a);
   }
+  for (const anchor of document.querySelectorAll('.markup [dir="auto"] > a[href^="#"][rel="nofollow"]')) {
+    const originalId = anchor.getAttribute('href').replace(/^#user-content-/, '');
```

Comment:
Check first if the link matches the regexp and if it does not `continue` the loop.